integration-testing/src/main/proto/io/grpc/testing/integration/test.proto

```diff
@@ -58,6 +58,11 @@ service TestService {
   rpc StreamingInputCall(stream StreamingInputCallRequest)
       returns (StreamingInputCallResponse);
 
+  // Start uploading only after receiving a confirmation message from the server.
+  // The server returns the aggregated size of client payload as the final result.
+  rpc CheckedUploadCall(stream StreamingInputCallRequest)
```

Comment:
Yeah, I used bidi streaming because it can return a final response at last, which can be used to verify the correct data has been sent and received.

How about "CheckAndUploadCall"?